### ISSUE TITLE
.htaccess rules bypassed when cached url contains special characters

### ISSUE THREAD

**Issue by rbrv** [putyourlightson/craft-blitz#365]
- First visitor goes to https://example.com/news?filter=foo%2Cbar
- Page is not cached yet, Blitz caches as `web/cache/blitz/example.com/news/filter=foo,bar/index.html`
- Second visitor goes to https://example.com/news?filter=foo%2Cbar
- Because the comma is encoded in the url as `%2C`, the apache rewrite does not match and will not serve the cached page

Using the Apache `.htaccess` code as found on https://putyourlightson.com/plugins/blitz#apache

**Comment by bencroker** [comment]
Blitz should be creating a cached file at the URL decoded path too. Can you ensure that you're running the latest version of the plugin?

**Comment by bencroker** [comment]
Blitz should be creating a cached file at the unencoded path too. Can you ensure that you're running the latest version of Blitz?

**Comment by rbrv** [comment]
Blitz only creates the unencoded path `web/cache/blitz/example.com/news/filter=foo,bar/index.html` (the `,` between `foo` and `bar`. 

We are running the latest version; `3.11.1`.

`.htaccess` rules:
```htaccess
# Asset cache busting
RewriteCond %{REQUEST_FILENAME} !-f
RewriteRule ^(.+)\.(?:\d+)\.(js|css|png|jpg|jpeg|gif|svg|webp)$ $1.$2 [L]

#- Blitz cache rewrite
RewriteCond %{DOCUMENT_ROOT}/cache/blitz/%{HTTP_HOST}/%{REQUEST_URI}/%{QUERY_STRING}/index.html -s
RewriteCond %{REQUEST_METHOD} GET
RewriteCond %{QUERY_STRING} !token= [NC]
RewriteRule .* /cache/blitz/%{HTTP_HOST}/%{REQUEST_URI}/%{QUERY_STRING}/index.html [L]

# Send would-be 404 requests to Craft
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteCond %{REQUEST_URI} !^/(favicon\.ico|apple-touch-icon.*\.png)$ [NC]
RewriteRule (.+) index.php/$1 [QSA,L]
```

**Comment by bencroker** [comment]
Thanks for clarifying, we'll look into a fix for this.

**Comment by bencroker** [comment]
Just pushed a fix in https://github.com/putyourlightson/craft-blitz/commit/9915f9c25bf2b2885695ab74b9d5855689eb2ec7 which should resolve this issue. Feel free to test and let me know if it works for you.

**Comment by bencroker** [comment]
Released in version 3.11.2.

**Comment by rbrv** [comment]
Only just had the possibility to update. I can indeed confirm it's working now. Thanks @bencroker !
